cmake/information_helpers.cmake

```diff
@@ -10,6 +10,8 @@ function(filter_generator_expressions INPUT OUTPUT)
     string(REGEX REPLACE "[$<A-Z_:]*SHELL:(.+)>*" "\\1" TMP "${TMP}")
     string(REGEX REPLACE ".+INTERFACE:.+>" "" TMP "${TMP}")
     string(REGEX REPLACE "\$<COMMA>" "," TMP "${TMP}")
+    # Remove the left : or > 
+    string(REGEX REPLACE ":|>" "" TMP "${TMP}")
```

Comment:
I don't like the amount of complexity we're collecting here. I think we should consider using CMake's features to evaluate the generator expressions if possible.

Reply:
Could you check if `file(GENERATE ...)` already evaluates the expressions? It should according to [the documentation](https://cmake.org/cmake/help/latest/command/file.html#generate), but some expressions have scopes which may not be available in this step.

Reply:
I already checked in the beginning. It requires CMake 3.15 to support TARGET. `file(GENERATE ... TARGET ginkgo)` uses the property of ginkgo to generate the file. Moreover, it generates the file what we want but it throw the error about multiple modification are different in the file. I do not dig into the issue yet

Reply:
Seems to be even 3.19, so this would be a pretty big jump. I guess we can deal with the current approach for a while longer, maybe add a TODO to remove this once we bump the version requirements